polaris-react/src/components/Columns/Columns.tsx

```diff
@@ -50,10 +60,43 @@ export function Columns({children, columns, gap = '4'}: ColumnsProps) {
   );
 }
 
-function formatColumns(columns?: number | string) {
+function formatColumns(columns?: Columns): ResponsiveValue {
+  if (
+    typeof columns === 'object' &&
+    columns !== null &&
+    !Array.isArray(columns)
+  ) {
+    return Object.fromEntries(
+      Object.entries(columns).map(([breakpointAlias, breakpointColumns]) => [
+        breakpointAlias,
+        getColumnValue(breakpointColumns),
+      ]),
+    );
+  }
+
+  return getColumnValue(columns);
+}
+
+function getColumnValue(columns?: ColumnTypes) {
   if (!columns) return undefined;
 
-  return typeof columns === 'number'
-    ? `repeat(${columns}, minmax(0, 1fr))`
-    : columns;
+  if (typeof columns === 'string') return columns;
+
+  if (typeof columns === 'number') {
+    return `repeat(${columns}, minmax(0, 1fr))`;
+  }
+
+  return columns
+    .map((column) => {
+      switch (column) {
+        case 'oneThird':
+        case 'oneHalf':
```

Comment:
I find it a little weird to have a case without a return. Can we do something like:

```suggestion
        case 'oneHalf' || 'oneThird':
```

Reply:
That doesn't work.

Both cases return the same thing. Could do this if it needs to be more clear.

```
case 'oneThird':
  return 'minmax(0, 1fr)';
case 'oneHalf':
  return 'minmax(0, 1fr)';
```

Reply:
I like that 👍 

Do we also need a full width do you think?

Reply:
I don't think so. The value in the aliases is that it adds `minmax()` for free. But I don't think there's a benefit to it if there's only 1 column.